yew-macro/src/html_classes/mod.rs

```diff
@@ -0,0 +1,34 @@
+use proc_macro2::TokenStream;
+use quote::{quote, ToTokens};
+use syn::parse::{Parse, ParseStream, Result};
+use syn::punctuated::Punctuated;
+use syn::token::Comma;
+use syn::Expr;
+
+/// List of HTML classes.
+pub struct HtmlClasses(Punctuated<Expr, Comma>);
+
+impl Parse for HtmlClasses {
+    fn parse(input: ParseStream) -> Result<Self> {
+        Ok(HtmlClasses(Punctuated::<Expr, Comma>::parse_terminated(
+            input,
+        )?))
+    }
+}
+
+impl ToTokens for HtmlClasses {
```

Comment:
To give the `classes!` macro a bit more of an edge (and justify it being a proc macro instead of a declarative macro) I would like is to add special treatment for string literals.
Most `classes!` invocations will probably involve at least one string literal and that way we can optimize them:
- String literals containing a space should error with the suggestion of splitting them apart (`classes!("hello world")` should be written as `classes!("hello", "world")`, because with the latter it's much clearer what's going on).
- They should use an unsafe `unchecked_push` method that adds a class to the `Classes` value without having to split it up first. The compiler *might* be smart enough to do this already, but by doing it explicitly we can be sure it's happening and hopefully even cut down on compile times. 

Do you think you could do that?

Reply:
Fixed in e748001 (ensure not space in class name)

Fixed in c30ee6c (use `unchecked_push()` for string literals)

Reply:
```
error: string literals should not contain spaces: please use two separate literals
   --> yew/src/virtual_dom/classes.rs:303:44
    |
303 |             <div class=classes!("class-1", "class-1 class-2")></div>
    |                                            ^^^^^^^^^^^^^^^^^

error: string literals should not contain spaces: please use two separate literals
   --> yew/src/virtual_dom/classes.rs:318:33
    |
318 |             <div class=classes!("class-1 class-2 class-3")></div>
    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^

error: string literals should not contain spaces: please use two separate literals
   --> yew/src/virtual_dom/classes.rs:321:33
    |
321 |             <div class=classes!("class-2 class-3 class-1")></div>
    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^
```

I guess it worked :joy: Now I need to fix the other tests